pkg/build/builder/common.go

```diff
@@ -180,3 +155,47 @@ func execPostCommitHook(client DockerClient, postCommitSpec api.BuildPostCommitS
 		Stderr:       true,
 	})
 }
+
+func updateBuildRevision(build *api.Build, sourceInfo *git.SourceInfo) *api.SourceRevision {
+	if build.Spec.Revision != nil {
+		return build.Spec.Revision
+	}
+	return &api.SourceRevision{
+		Git: &api.GitSourceRevision{
+			Commit:  sourceInfo.CommitID,
+			Message: sourceInfo.Message,
+			Author: api.SourceControlUser{
+				Name:  sourceInfo.AuthorName,
+				Email: sourceInfo.AuthorEmail,
+			},
+			Committer: api.SourceControlUser{
+				Name:  sourceInfo.CommitterName,
+				Email: sourceInfo.CommitterEmail,
+			},
+		},
+	}
+}
+
+func updateStatusRetrier(build *api.Build, client client.BuildInterface, sourceRev *api.SourceRevision) error {
+	return kclient.RetryOnConflict(kclient.DefaultBackoff, func() error {
```

Comment:
@smarterclayton can you weigh back in?  is using patch from controllers(or in this case, a process that's subservient to a controller) a pattern we're trying to move to?  or should this be left as is since it's aligned w/ what we're doing elsewhere today?

@kargakis can you point us to the code where you do patch in the idling controller?